### PR TITLE
Page layout metrics

### PR DESCRIPTION
**Pull request by kaero598**
The missing `Page.getLayoutMetrics` feature is the only thing that prevents taking full-size screenshots.

### CODE REVIEW

**Comment by gsouf** [comment]
Hi @kaero598,
that's a great first contribution, thanks!

I'll just add one more scenario in unit tests because you're testing the window size, but you don't test the case where the layout is larger than the window.

**Comment by gsouf** [comment]
@kaero598 I added a comment, I'll wait for your response before I go ahead

**Comment by kaero598** [comment]
@gsouf Nice catch. I made #45 covering this scenario.

**Comment by gsouf** [comment]
@kaero598 Thanks, but I did it already, but didn't push yet as I was waiting for your answer about the LayoutMetricFailed excption

**Comment by kaero598** [comment]
Nope. I see code changes and no comments. Your link just highlights the line before exception is thrown.

**Comment by gsouf** [comment]
Sorry I never used that new review tool. I guess you can see it now

File: src/PageUtils/PageLayoutMetrics.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        $this->responseReader = $responseReader;
+    }
+
+    public function waitForResponse()
+    {
+        $this->response = $this->responseReader->waitForResponse();
+
+        if (!$this->response->isSuccessful()) {
+            throw new LayoutMetricsFailed('Could not retrieve layout metrics of the page.');
```
[the remark this answers is not in the document]
Comment (kaero598) [reply src/PageUtils/PageLayoutMetrics.php]: There were no specific reasons. I mimicked behavior of `evaluate()`.

[the remark this answers is not in the document]
Comment (kaero598) [reply src/PageUtils/PageLayoutMetrics.php]: Okay. I see your point.

**Pull request merged**